exonum-java-binding-core/src/main/java/com/exonum/binding/service/Service.java

```diff
@@ -118,4 +118,13 @@
    * @param router a router responsible for handling requests to this service
    */
   void createPublicApiHandlers(Node node, Router router);
+
+  /**
+   * Handles block commit event. This handler is a callback method which is invoked by
+   * the blockchain after each block commit.
+   * For example, a service can create one or more transactions if a specific condition
+   * has occurred.
```

Comment:
Please add as a separate paragraph a warning that this method is invoked synchronously from the thread that commits the block, therefore, implementations of this method must not perform any blocking or long-running operations.